### DISCUSSION TITLE
The Hash - Dojo webhooks examples

### DISCUSSION THREAD

**Discussion by gyroscopes** [dojo-engineering/feedback#8] [Docs]
On this page
https://docs.dojo.tech/docs/development-resources/webhooks

Step 3. Verify Dojo webhooks

Here is this json

```
{
  "id": "evt_hnnHxIKR_Uy6bhZCusCltw",
  "event": "payment_intent.created", 
  "accountId": "acc_test", 
  "createdAt": "2022-02-01T13:07:41.8667859Z",
  "data": { 
    "paymentIntentId": "pi_vpwd4ooAPEqyNAQe4z89WQ", 
    "paymentStatus": "Created", 
    "captureMode": "Auto"
    }
}
```

and with this secret  `ws_OYsyzcrH70GpJfs8MMkKpg`
once HMAC-SHA256 hashed it is claimed should get 
`sha256=94-AC-93-67-D6-8B-69-6D-15-A2-C2-45-40-98-B5-E7-64-72-38-C5-1F-6D-32-D0-5F-CB-91-20-A9-20-64-F4`

But using online HMAC-SHA256 checkers I'm getting
`4c8febded69a1ea33530ac7a5706bf840b4a5d24fd4b715c9a5e64377d1bf19c`

I'm planning on figuring out why they are different today.

**Comment by gyroscopes** [comment]
I've done another test.
I opened 
https://github.com/dojo-engineering/dojo-samples/tree/main/samples/accept-a-payment/prebuilt-checkout/server)/server.csproj
in visual studio (I found this project compiles straight away with out any editing).

I then add a single line (rather long line) in program.cs

`var payload = WebhookPayloadUtils.ReadPayload("{\r\n  \"id\": \"evt_hnnHxIKR_Uy6bhZCusCltw\",\r\n  \"event\": \"payment_intent.created\", \r\n  \"accountId\": \"acc_test\", \r\n  \"createdAt\": \"2022-02-01T13:07:41.8667859Z\",\r\n  \"data\": { \r\n    \"paymentIntentId\": \"pi_vpwd4ooAPEqyNAQe4z89WQ\", \r\n    \"paymentStatus\": \"Created\", \r\n    \"captureMode\": \"Auto\"\r\n    }\r\n}", "ws_OYsyzcrH70GpJfs8MMkKpg", "sha256=94-AC-93-67-D6-8B-69-6D-15-A2-C2-45-40-98-B5-E7-64-72-38-C5-1F-6D-32-D0-5F-CB-91-20-A9-20-64-F4");`

This will cause an exception if the hash is incorrect. I cut and paste the sample json carefully (see above) along with the secret. 
Sure enough it caused an exception.

Then took the code directly from Dojo.Net.Webhooks.WebbhookPayloadUtils 
and created my own simple test function but with the most minimum changes.
Sure enough I got the following hash.

`4C-8F-EB-DE-D6-9A-1E-A3-35-30-AC-7A-57-06-BF-84-0B-4A-5D-24-FD-4B-71-5C-9A-5E-64-37-7D-1B-F1-9C`

So either I'm not handling/formatting the data before it goes into the functions or there is minor issue with the json data. 

Maybe a space when there shouldn't be. Maybe worth checking.

**Reply by GabrielBoceanuDojo** [reply]
Hi Glenn,

Thank for not only looking at our docs but for also going to such extent to verify and validate the correctness of our sample codes and provide us valuable feedback.
I had a look as well, I also can confirm that we potentially have some misalignment in the sample.
Would you be able to attempt to test the code again with the following signature instead: **sha256=E0-EA-40-35-68-0B-61-A2-D7-A8-CE-92-BB-64-36-0B-96-11-9F-EA-17-37-34-B8-EE-81-4A-0D-41-08-E8-70**

We will update the documentation as well to reflect this as soon as possible.

Thanks,
Gabriel Boceanu

**Reply by RomanKondratyev** [reply]
I have updated the secret (to reflect latest look of it) and put another calculated secret, should match now.
